shims/spark35/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala

```diff
@@ -0,0 +1,241 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.execution
```

Comment:
It seems this Spark object is introduced, could you please also make a note what is the required change here?

Reply:
@zhztheplayer @FelixYBW 
This is indeed a bug affecting Spark versions 3.5 and 4.0. The Spark community is supportive of this fix and is preparing to merge it into the main branch. Once merged, I will backport the fix to Spark 3.5. Thanks.

Reply:
Even it's merged, we still need to back port to 4.0, right?

Reply:
Since it's an important spark file. 

Can we comment somehow in the file which lines of code are changes? Or most of the files are changed?

Reply:
I meant we are free to pass the Spark context to the listener so avoid this replacement. No? @JkSelf 

> The Spark community is supportive of this fix and is preparing to merge it into the main branch. 

Yes it's reasonable to move to a vanilla solution but before that let's check whether we have to apply such a change.